### PR TITLE
[enhance](auth)when assigning permissions, the current user must have corresponding permissions

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

When authorizing other users or roles, not only need `grant_priv`,  also need the priv want to grant
for example ：
The user executes the following SQL
```
grant select_priv,load_priv on db1.table1 to u1;
```
he need have grant_priv, select_priv and load_priv on db1.table1

```
grant usage_priv on resource r1 to u1;
```
he need to have grant_priv and usage_priv on u1

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/GrantStmt.java [hunk 1]
```diff
@@ -37,7 +39,9 @@
 import com.google.common.collect.Sets;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.commons.collections.MapUtils;
+import org.apache.hadoop.util.Lists;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/GrantStmt.java:42]: Wrong import

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Review by BePPPower — APPROVED** [review]

**Pull request merged**